### ISSUE TITLE
Issue with latest version

### ISSUE THREAD

**Issue by perimore** [aristanetworks/openconfigbeat#2]
Hi,

I tried the latest version of the beat and ran into this issue related to templates:

```
arista@lanzLab-vm2:~/go/bin$ ./openconfigbeat -e -d "1"
2016/11/08 21:49:53.685657 beat.go:223: INFO Home path: [/home/arista/go/bin] Config path: [/home/arista/go/bin] Data path: [/home/arista/go/bin/data] Logs path: [/home/arista/go/bin/logs]
2016/11/08 21:49:53.685757 beat.go:244: INFO Setup Beat: openconfigbeat; Version: 5.0.0-SNAPSHOT
2016/11/08 21:49:53.685829 output.go:141: INFO Loading template enabled. Reading template file: /home/arista/go/bin/openconfigbeat.template.json
2016/11/08 21:49:53.685919 outputs.go:88: INFO Activated elasticsearch as output plugin.
2016/11/08 21:49:53.686021 publish.go:292: INFO Publisher name: lanzLab-vm2
2016/11/08 21:49:53.686177 async.go:63: INFO Flush Interval set to: 1s
2016/11/08 21:49:53.686205 async.go:64: INFO Max Bulk Size set to: 50
2016/11/08 21:49:53.686456 beat.go:272: INFO openconfigbeat start running.
2016/11/08 21:49:53.686484 openconfigbeat.go:124: INFO openconfigbeat is running! Hit CTRL-C to stop it.
2016/11/08 21:49:53.686607 openconfigbeat.go:133: INFO Connected to 172.28.130.120:6042
2016/11/08 21:49:53.687860 openconfigbeat.go:133: INFO Connected to 172.28.129.109:6042
2016/11/08 21:49:53.688671 openconfigbeat.go:163: INFO Sending subscribe request: subscribe:<subscription:<path:<element:"" element:"LANZ" > > >
2016/11/08 21:49:53.688824 openconfigbeat.go:163: INFO Sending subscribe request: subscribe:<subscription:<path:<element:"" element:"LANZ" > > >
2016/11/08 21:49:53.688952 openconfigbeat.go:163: INFO Sending subscribe request: subscribe:<subscription:<path:<element:"" element:"Sysdb" element:"environment" element:"temperature" element:"status" > > >
2016/11/08 21:49:53.689083 openconfigbeat.go:163: INFO Sending subscribe request: subscribe:<subscription:<path:<element:"" element:"Sysdb" element:"environment" element:"temperature" element:"status" > > >
2016/11/08 21:49:53.689324 openconfigbeat.go:163: INFO Sending subscribe request: subscribe:<subscription:<path:<element:"" element:"Sysdb" element:"environment" element:"power" element:"status" > > >
2016/11/08 21:49:53.689530 openconfigbeat.go:163: INFO Sending subscribe request: subscribe:<subscription:<path:<element:"" element:"Sysdb" element:"environment" element:"power" element:"status" > > >
2016/11/08 21:49:53.689741 openconfigbeat.go:163: INFO Sending subscribe request: subscribe:<subscription:<path:<element:"" element:"Sysdb" element:"environment" element:"thermostat" element:"status" > > >
2016/11/08 21:49:53.690073 openconfigbeat.go:163: INFO Sending subscribe request: subscribe:<subscription:<path:<element:"" element:"Sysdb" element:"environment" element:"thermostat" element:"status" > > >
2016/11/08 21:49:53.690660 openconfigbeat.go:163: INFO Sending subscribe request: subscribe:<subscription:<path:<element:"" element:"Sysdb" element:"environment" element:"cooling" element:"status" > > >
2016/11/08 21:49:53.691156 openconfigbeat.go:163: INFO Sending subscribe request: subscribe:<subscription:<path:<element:"" element:"Sysdb" element:"environment" element:"cooling" element:"status" > > >
2016/11/08 21:49:53.702771 output.go:159: INFO Trying to load template for client: &{{http://localhost:9200   %!s(*http.Client=&{0xc4201d4000 <nil> <nil> 90000000000}) %!s(bool=true) %!s(func() error=0x5e9140)} openconfigbeat map[] {{ <nil> %!s(bool=false) %!s(int=0) %!s(int=0) %!s(int=0)}  %!s(elasticsearch.state=0) }}
2016/11/08 21:49:53.703811 output.go:166: INFO Existing template will be overwritten, as overwrite is enabled.
2016/11/08 21:49:54.665330 single.go:130: INFO Connecting error publishing events (retrying): Connection marked as failed because the onConnect callback failed: Could not load template: Template could not be loaded. Error: 400 Bad Request
2016/11/08 21:49:54.665360 single.go:145: INFO send fail
2016/11/08 21:49:55.666829 output.go:159: INFO Trying to load template for client: &{{http://localhost:9200   %!s(*http.Client=&{0xc4201d4000 <nil> <nil> 90000000000}) %!s(bool=true) %!s(func() error=0x5e9140)} openconfigbeat map[] {{ <nil> %!s(bool=false) %!s(int=0) %!s(int=0) %!s(int=0)}  %!s(elasticsearch.state=0) }}
2016/11/08 21:49:55.667460 output.go:166: INFO Existing template will be overwritten, as overwrite is enabled.
**2016/11/08 21:49:56.288497 single.go:130: INFO Connecting error publishing events (retrying): Connection marked as failed because the onConnect callback failed: Could not load template: Template could not be loaded. Error: 400 Bad Request**
2016/11/08 21:49:56.288522 single.go:145: INFO send fail
^C2016/11/08 21:49:57.440841 beat.go:280: INFO openconfigbeat cleanup
```

Any other info needed? I tried an older version which gets further but each stat is within a new branch of JSON, i.e.:

```
"update": {
      "Sysdb": {
        "interface": {
          "counter": {
            "eth": {
              "slice": {
                "phy": {
                  "Linecard3": {
                    "intfCounterDir": {
                      "Ethernet3/13/1": {
                        "intfCounter": {
                          "current": {
                            "rates": {
                              "outBitsRate": 617.4555192973619,
                              "rates": {
                                "inBitsRate": 5e-324,
                                "rates": {
                                  "rates": {
                                    "outPktsRate": 0.680042492919596,
                                    "rates": {
                                      "ethStatistics": {
                                        "ethStatistics": {
                                          "ethStatistics": {
                                            "ethStatistics": {
                                              "ethStatistics": {
                                                "ethStatistics": {
                                                  "ethStatistics": {
                                                    "ethStatistics": {
                                                      "ethStatistics": {
                                                        "ethStatistics": {
                                                          "ethStatistics": {
                                                            "ethStatistics": {
                                                              "ethStatistics": {
                                                                "ethStatistics": {
                                                                  "ethStatistics": {
                                                                    "ethStatistics": {
                                                                      "ethStatistics": {
                                                                        "ethStatistics": {
                                                                          "ethStatistics": {
                                                                            "ethStatistics": {
                                                                              "ethStatistics": {
                                                                                "ethStatistics": {
                                                                                  "carrierSenseErrors": 0,
                                                                                  "ethStatistics": {
                                                                                    "ethStatistics": {
                                                                                      "ethStatistics": {
                                                                                        "ethStatistics": {
                                                                                          "ethStatistics": {
                                                                                            "ethStatistics": {
                                                                                              "ethStatistics": {
                                                                                                "ethStatistics": {
                                                                                                  "ethStatistics": {
                                                                                                    "deferredTransmissions": 0,
                                                                                                    "ethStatistics": {
                                                                                                      "alignmentErrors": 0,
                                                                                                      "ethStatistics": {
                                                                                                        "ethStatistics": {
                                                                                                          "ethStatistics": {
                                                                                                            "ethStatistics": {
                                                                                                              "ethStatistics": {
                                                                                                                "singleCollisionFrames": 0,
                                                                                                                "statistics": {
                                                                                                                  "inMulticastPkts": 0,
                                                                                                                  "statistics": {
                                                                                                                    "outOctets": 1535768817742,
                                                                                                                    "statistics": {
                                                                                                                      "inOctets": 11072,
                                                                                                                      "statistics": {
                                                                                                                        "lastUpdate": 1634419.025426643,
                                                                                                                        "statistics": {
                                                                                                                          "outMulticastPkts": 1185161,
                                                                                                                          "statistics": {
                                                                                                                            "outBroadcastPkts": 143,
                                                                                                                            "statistics": {
                                                                                                                              "inUcastPkts": 107,
                                                                                                                              "statistics": {
                                                                                                                                "inErrors": 0,
                                                                                                                                "statistics": {
                                                                                                                                  "outDiscards": 5793114,
                                                                                                                                  "statistics": {
                                                                                                                                    "outErrors": 0,
                                                                                                                                    "statistics": {
                                                                                                                                      "outUcastPkts": 1011614269,
                                                                                                                                      "statistics": {
                                                                                                                                        "inBroadcastPkts": 66,
                                                                                                                                        "statistics": {
                                                                                                                                          "inDiscards": 0
                                                                                                                                        }
                                                                                                                                      }
                                                                                                                                    }
                                                                                                                                  }
                                                                                                                                }
                                                                                                                              }
                                                                                                                            }
                                                                                                                          }
                                                                                                                        }
                                                                                                                      }
                                                                                                                    }
                                                                                                                  }
                                                                                                                }
                                                                                                              },
                                                                                                              "out64OctetFrames": 513
                                                                                                            },
                                                                                                            "frameTooLongs": 0
                                                                                                          },
                                                                                                          "frameTooShorts": 0
                                                                                                        },
                                                                                                        "internalMacTransmitErrors": 0
                                                                                                      }
                                                                                                    }
                                                                                                  },
                                                                                                  "inPfcFrames": 0
                                                                                                },
                                                                                                "outPfcFrames": 0
                                                                                              },
                                                                                              "in128To255OctetFrames": 0
                                                                                            },
                                                                                            "excessiveCollisions": 0
                                                                                          },
                                                                                          "multipleCollisionFrames": 0
                                                                                        },
                                                                                        "in1523ToMaxOctetFrames": 0
                                                                                      },
                                                                                      "inPauseFrames": 0
                                                                                    },
                                                                                    "inUnknownOpcodes": 0
                                                                                  }
                                                                                },
                                                                                "inPfcClassFrames": [
                                                                                  0,
                                                                                  0,
                                                                                  0,
                                                                                  0,
                                                                                  0,
                                                                                  0,
                                                                                  0,
                                                                                  0
                                                                                ]
                                                                              },
                                                                              "out1523ToMaxOctetFrames": 0
                                                                            },
                                                                            "out256To511OctetFrames": 0
                                                                          },
                                                                          "out65To127OctetFrames": 1130699
                                                                        },
                                                                        "outPauseFrames": 0
                                                                      },
                                                                      "out512To1023OctetFrames": 0
                                                                    },
                                                                    "fcsErrors": 0
                                                                  },
                                                                  "out1024To1522OctetFrames": 1011613899
                                                                },
                                                                "in512To1023OctetFrames": 0
                                                              },
                                                              "jabbers": 0
                                                            },
                                                            "symbolErrors": 0
                                                          },
                                                          "internalMacReceiveErrors": 0
                                                        },
                                                        "lateCollisions": 0
                                                      },
                                                      "in64OctetFrames": 173
                                                    },
                                                    "outPfcClassFrames": [
                                                      0,
                                                      0,
                                                      0,
                                                      0,
                                                      0,
                                                      0,
                                                      0,
                                                      0
                                                    ]
                                                  },
                                                  "in65To127OctetFrames": 0
                                                },
                                                "sqeTestErrors": 0
                                              },
                                              "out128To255OctetFrames": 54462
                                            },
                                            "in1024To1522OctetFrames": 0
                                          },
                                          "in256To511OctetFrames": 0
                                        },
                                        "fragments": 0
                                      },
                                      "inPktsRate": 5e-324
                                    }
                                  },
                                  "statsUpdateTime": 1634419.025426643
                                }
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  },
  "fields": {
    "@timestamp": [
      1478641466980
    ]
  },
  "sort": [
    1478641466980
  ]
}
```

**Comment by tsuna** [comment]
We've upgraded everything to work with the latest elasticsearch / logstash etc, and fixed a bug that caused the funny looking crazy-nested JSON you posted above.  Can you confirm that everything now works OK for you?

**Comment by 7AC** [comment]
15c770a fixes the template issue, the JSON issue should be fixed by aristanetworks/goarista@8982df8
